### PR TITLE
Add further summaries

### PR DESCRIPTION
**Pull request by shainaraskas**
This PR adds operation summaries for the following areas, using text from the appropriate pages from the [elasticsearch API reference](https://www.elastic.co/guide/en/elasticsearch/reference/current/rest-apis.html).

- CAT
- doc
- delete by query
- update by query
- enrich
- info
- mapping
- transform
- validate

Also converted the messy notes into note blocks, e.g. 

image

part of https://github.com/elastic/elasticsearch-specification/issues/2635

will regenerate the reference after https://github.com/elastic/elasticsearch-specification/pull/2718 is approved and merged and this PR is rebased

### CODE REVIEW

File: specification/cat/indices/CatIndicesRequest.ts [hunk 1]
```diff
@@ -22,12 +22,21 @@ import { Bytes, ExpandWildcards, HealthStatus, Indices } from '@_types/common'
...
 /**
+ * Get index information.
  * Returns high-level information about indices in a cluster, including backing indices for data streams.
- * IMPORTANT: cat APIs are only intended for human consumption using the command line or Kibana console.
- * They are not intended for use by applications. For application consumption, use the get index API.
- * Use the cat indices API to get the following information for each index in a cluster: shard count; document count; deleted document count; primary store size; total store size of all shards, including shard replicas.
+ * > info
+ * > CAT APIs are only intended for human consumption using the command line or Kibana console.
+ * They are not intended for use by applications. For application consumption, use an index endpoint.
+ *
```
Comment (shainaraskas) [inline specification/cat/indices/CatIndicesRequest.ts:30]: these empty lines are failing CI, but newline characters (\n) are still being escaped. investigating

Comment (shainaraskas) [reply specification/cat/indices/CatIndicesRequest.ts:30]: played around a little with this. `<br>` is just adding newlines into the note block, not "ending" the note block: 

```
        "description": "Returns high-level information about indices in a cluster, including backing indices for data streams.\n> info\n> CAT APIs are only intended for human consumption using the command line or Kibana console.\nThey are not intended for use by applications. For application consumption, use an index endpoint.<br><br>Use this request to get the following information for each index in a cluster:\n- shard count\n- document count\n- deleted document count\n- primary store size\n- total store size of all shards, including shard replicas\n\nThese metrics are retrieved directly from Lucene, which Elasticsearch uses internally to power indexing and search. As a result, all document counts include hidden nested documents.\nTo get an accurate count of Elasticsearch documents, use the [/_cat/count](#operation-cat-count) or [count](#endpoint-count) endpoints.",
```

image

explicit newline characters are being escaped:

```
        "description": "Returns high-level information about indices in a cluster, including backing indices for data streams.\n> info\n> CAT APIs are only intended for human consumption using the command line or Kibana console.\nThey are not intended for use by applications. For application consumption, use an index endpoint.\\n\nUse this request to get the following information for each index in a cluster:\n- shard count\n- document count\n- deleted document count\n- primary store size\n- total store size of all shards, including shard replicas\\n\nThese metrics are retrieved directly from Lucene, which Elasticsearch uses internally to power indexing and search. As a result, all document counts include hidden nested documents.\nTo get an accurate count of Elasticsearch documents, use the [/_cat/count](#operation-cat-count) or [count](#endpoint-count) endpoints.",
```

image

and explicit spaces are being parsed as `\n\n` into the reference, but failing CI.

image

File: specification/indices/validate_query/IndicesValidateQueryRequest.ts [hunk 2]
```diff
@@ -23,7 +23,8 @@ import { QueryContainer } from '@_types/query_dsl/abstractions'
 import { Operator } from '@_types/query_dsl/Operator'
 
 /**
- * Validates a potentially expensive query without executing it.
+ * Validate a query.
+ * Validates a query without executing it.
```
Comment (lcawl) [inline specification/indices/validate_query/IndicesValidateQueryRequest.ts:27]:
```suggestion
 * Validates a query without running it.
```

**Review by lcawl — APPROVED** [review]
Cool callout additions, LGTM

**Pull request merged**